motoman_es_support/urdf/es165rdii_macro.xacro

```diff
@@ -154,56 +162,62 @@
         <material name="yellow"/>
       </collision>
     </link>
+    -->
     <!-- end of link list -->
 
     <!-- joint list -->
-    <joint name="${prefix}joint_s" type="revolute">
+    <joint name="${prefix}joint_1_s" type="revolute">
       <origin xyz="0 0 0.450" rpy="0 0 0"/>
       <parent link="${prefix}base_link"/>
       <child link="${prefix}link_s"/>
       <axis xyz="0 0 1"/>
-      <limit lower="${-radians(180)}" upper="${radians(180)}" effort="0" velocity="${radians(105)}"/>
+      <limit lower="${radians(-180)}" upper="${radians(180)}" effort="0" velocity="${radians(105)}"/>
     </joint>
-    <joint name="${prefix}joint_l" type="revolute">
+    <joint name="${prefix}joint_2_l" type="revolute">
       <origin xyz="0.740 0 0" rpy="0 0 0"/>
       <parent link="${prefix}link_s"/>
       <child link="${prefix}link_l"/>
       <axis xyz="0 1 0"/>
-      <limit lower="${-radians(130)}" upper="${radians(80)}" effort="0" velocity="${radians(105)}"/>
+      <limit lower="${radians(-130)}" upper="${radians(80)}" effort="0" velocity="${radians(105)}"/>
     </joint>
-    <joint name="${prefix}joint_u" type="revolute">
+    <joint name="${prefix}joint_3_u" type="revolute">
       <origin xyz="1.150 0 0" rpy="0 0 0"/>
       <parent link="${prefix}link_l"/>
       <child link="${prefix}link_u"/>
       <axis xyz="0 -1 0"/>
-      <limit lower="${-radians(138.0)}" upper="${radians(78.0)}" effort="0" velocity="${radians(105)}"/>
+      <limit lower="${radians(-112)}" upper="${radians(208)}" effort="0" velocity="${radians(105)}"/>
```

Comment:
Do we want to represent joint limits like this one (whose limits are not defined relative to the previous joint) verbatim per the spec sheet? In the case of this particular joint, I believe its limits change based on the value of joint L and are measured relative to a fixed vertical plane. We cannot represent this with URDF, but the [-138, 78] relative range more accurately (but not totally accurately) represents the physical range of the robot

Reply:
I changed this as we use the specced joint limits for all other robot support packages, across all supported vendors.

Doing something different here would mean diverging from that, which would require additional documentation.

We can do this, but I'd like to do it only if necessary.

Perhaps @EricMarcil and/or @ted-miller have an idea about what would be best to do here?

Reply:
This joint's limit does not change based on the value of L. It is measured relative to L and not to a vertical plane.
(There some some 4-axis palletizing robots which are fixed on a vertical plane. But those are mechanically fixed in that orientation.)
![Annotation 2020-07-30 145834](https://user-images.githubusercontent.com/4623811/88963945-99692480-d276-11ea-9c12-6c4f74c4daac.png)
![Annotation 2020-07-30 150359](https://user-images.githubusercontent.com/4623811/88963949-9bcb7e80-d276-11ea-804f-5e4a91c3b2ba.png)
![Annotation 2020-07-30 151643](https://user-images.githubusercontent.com/4623811/88964762-d41f8c80-d277-11ea-9067-646160ef0e72.png)
![Annotation 2020-07-30 151707](https://user-images.githubusercontent.com/4623811/88964764-d5e95000-d277-11ea-8355-1af2a0cd3e2d.png)

Reply:
I haven't come across those limits in any of the spec sheets that I have for our 200R*-120 model, but it seems to make sense in the context of the work envelope diagram.

The robot in our lab definitely has a higher positive joint limit than the 79 degrees stated in the document above and can actually achieve an "elbow-down" configuration like in the screenshot below. 

![image](https://user-images.githubusercontent.com/18411310/88981011-94ae6b80-d28a-11ea-8614-f80c515dcbdb.png)

@ted-miller is this a non-standard feature? (I think you and @EricMarcil might have changed that configuration for us a long time ago) Or is this capability partially defined by the [-112 deg, 208 deg] joint limits on the spec sheet?

Reply:
I believe that your robot is a non-standard configuration. It is possible to modify the pulse limits for an axis. But, I expect that any modifications would impact the allowable moment and inertia for that axis. 
![Untitled](https://user-images.githubusercontent.com/4623811/89036385-961f7880-d30a-11ea-953b-8e0e5c08e750.png)

Reply:
Thanks for the info @ted-miller.

@marip8: this is actually why we always just use the specs in the datasheet / documentation.

It's perfectly possible for someone to configure different limits, but they'd have to do that in a fork.

Reply:
Okay, I'm on board with using the datasheet specs